core/domain/topic_domain.py

```diff
@@ -827,6 +889,24 @@ def update_name(self, new_name):
         """
         self.name = new_name
 
+    def update_abbreviated_name(self, new_abbreviated_name):
+        """Updates the abbreviated_name of a topic object.
+
+        Args:
+            new_abbreviated_name: str. The updated abbreviated_name
+            for the topic.
+        """
+        self.abbreviated_name = new_abbreviated_name
+
+    def update_thumbnail(self, new_thumbnail):
```

Comment:
Here and elsewhere I think we should be calling the "thing" thumbnail_filename rather than thumbnail. The latter refers to the image itself.

Reply:
Done.